app/models/calculator/weight.rb

```diff
@@ -66,10 +80,16 @@ def variant_unit(line_item)
       line_item.variant.product.andand.variant_unit
     end
 
-    def convert_g_to_kg(value)
+    def convert_weight(value)
       return 0 unless value
 
-      value / 1000
+      if preferences[:unit_from_list] == "kg"
+        value / 1000
+      elsif preferences[:unit_from_list] == "lb"
+        value / 453.6
+      else
+        raise "Unknown unit preference"
```

Comment:
maybe just this instead of else raise:
return 0 unless value && ["kg", "lb"].include?(value)

If you want to keep the raise you need to make it translatable :+1:

Reply:
Good point; I added the raise since I wanted to make sure it failed loudly if it were in an unexpected state, but since we have the validation, I think it's fine to remove the raise.